### PR TITLE
fix(cluster_controller): shift log error on cluster update to info level

### PR DESCRIPTION
**Pull request by dkropachev**
**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/scylladb/scylla-operator/issues/217

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [ ] Image has been built (`make docker-build`) on the last commit.

### CODE REVIEW

File: pkg/controllers/cluster/cluster_controller.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func (cc *ClusterReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
 	if !reflect.DeepEqual(c.Status, copy.Status) {
 		logger.Info(ctx, "Writing cluster status.")
 		if err := cc.Status().Update(ctx, copy); err != nil {
-			logger.Error(ctx, "Failed to update cluster status", "error", err)
+			logger.Info(ctx, "Failed to update cluster status", "error", err)
```
Comment (zimnx) [inline pkg/controllers/cluster/cluster_controller.go:124]: We want to log only conflicts on Info level. Other errors should be logged using Error.
You can use apierrors.IsConflict(err).

**Review by zimnx — CHANGES_REQUESTED** [review]

**Comment by zimnx** [comment]
Please also ammend commit to follow contributing guide: https://github.com/scylladb/scylla-operator/blob/master/docs/source/contributing.md#commit-messages

File: pkg/controllers/cluster/cluster_controller.go (hunk 1 above)
Comment (dkropachev) [reply pkg/controllers/cluster/cluster_controller.go]: fixed

**Comment by dkropachev** [comment]
> Please also ammend commit to follow contribution guide: https://github.com/scylladb/scylla-operator/blob/master/docs/source/contributing.md#commit-messages

Done

**Review by zimnx — APPROVED** [review]

**Pull request merged**